File: BlazorWindowHelper/BWHJsInterop.cs
using Microsoft.JSInterop;
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Components;

namespace BlazorWindowHelper
{
    public class BWHJsInterop
    {
        [Inject]
        static IJSRuntime JSRuntime { get; set; }

        //public static async Task<string> Prompt(string message)
        //{
           
        //    return await JSRuntime.InvokeAsync<string>(
        //        "BWHJsFunctions.showPrompt",
        //        message);
        //}

        //public static async Task<bool> Alert(string message)
        //{
        //    return await JSRuntime.InvokeAsync<bool>(
        //        "BWHJsFunctions.alert",message);
            
        //}

        //public static async Task<bool> Log(string message)
        //{
        //    return await JSRuntime.InvokeAsync<bool>(
        //        "BWHJsFunctions.log", message);
        //
        //}

        //public static async Task<bool> LogWithTime(string message)
        //{
        //    return await JSRuntime.InvokeAsync<bool>(
        //        "BWHJsFunctions.logWithTime", message);
        //
        //}

        public static async Task<bool> SetOnOrOff(bool OnOrOff)
        {
            return await JSRuntime.InvokeAsync<bool>(
                "BWHJsFunctions.setOnOrOff", OnOrOff);

        }
        
    }
}

File: BlazorWindowHelper/content/BWHJsInterop.js

var OnOrOff = false;

document.onkeyup = function (evt) {

    if (OnOrOff===true) {
        evt = evt || window.event;
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeKeyUp', evt.keyCode);
    }
};


window.addEventListener("scroll", onScroll, false);
window.addEventListener("resize", onResize, false);


function onScroll() {
    //if (OnOrOff === true) {
    console.log('BWHJsInterop: hit onScroll');
    DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeOnScroll');
    //}
}

function onResize() {
    if (OnOrOff === true) {
        console.log('BWHJsInterop: hit onResize');
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeOnResize');
    }
}


window.BWHJsFunctions = {
    showPrompt: function (message) {
    return prompt(message, 'Type anything here');
    },
    alert: function (message) {
        alert(message);
        return true;
    },
    log: function (message) {
        console.log(message);
        return true;
    },
    logWithTime: function (message) {
        console.log(getTime() +" " + message);
        return true;
    },
    setOnOrOff: function (b) {
        OnOrOff = b;
        return true;
    },
};
